File: sortAlgorithms/cBubbleSort/bubbleSort.c
#include <stdio.h>

void bubbleSort(int[], int);

void main() {
    int arrayNumber[] = {3, 9, 1, 5, 4, 7, 6, 8, 2};
    int arrayPositions = 9;
    printf("---Before Sort---\n");
    for(int i = 0; i < arrayPositions; i++)
    {
        printf("%d ", arrayNumber[i]);
    }
    
    bubbleSort(arrayNumber, arrayPositions);

    printf("\n---After Sort---\n");
    for(int i = 0; i < arrayPositions; i++)
    {
        printf("%d ", arrayNumber[i]);
    }
}

void bubbleSort(int arrayNumber[], int arrayPositions) {
    int swap;
    for(int i = 0; i < arrayPositions - 1; i++) 
    {
        for(int j = 0; j < arrayPositions - i - 1; j++) 
        {
            if(arrayNumber[j] > arrayNumber[j + 1]) {
                swap = arrayNumber[j];
                arrayNumber[j] = arrayNumber[j + 1];
                arrayNumber[j + 1] = swap;
            }
        }
    }
}

File: sortAlgorithms/javaBubbleSort/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {
    public static void main(String[] args) {
        int[] arrayNumbers = {8, 4, 1, 9, 7, 3, 6, 2, 5};
        System.out.println("---Before Sort---");
        Arrays.stream(arrayNumbers).forEach(value -> System.out.print(value + " "));
        bubbleSort(arrayNumbers);
        System.out.println("\n---After Sort---");
        Arrays.stream(arrayNumbers).forEach(value -> System.out.print(value +  " "));
    }

    private static void bubbleSort(int[] numbers) {
        int swap;
        for(int i = 0; i < numbers.length - 1; i++) {
            for(int j = 0; j < numbers.length - i - 1; j++) {
                if(numbers[j] > numbers[j + 1]) {
                    swap = numbers[j];
                    numbers[j] = numbers[j + 1];
                    numbers[j + 1] = swap;
                }
            }
        }
    }
}
